server/core/src/repl/mod.rs

```diff
@@ -363,11 +368,60 @@ async fn repl_run_consumer(
     Some(socket_addr)
 }
 
+#[instrument(level="info", skip(idms), fields(eventid=Uuid::new_v4().to_string()))]
+async fn replication_trigger(
+    idms: &IdmServer,
+    last_replication_instant: &Instant,
+    enable_experimental_replication_trigger: bool,
+    replication_interval: Duration,
+) {
+    tokio::select! {
+        _ = async {
+            let mut read_txn = idms.proxy_read().await;
+
+            let mut one_sec_int = interval(Duration::from_secs(1));
+            // sleep for the minimum window size before we do anything
+            sleep(Duration::from_secs(MIN_REPL_TASK_POLL_INTERVAL)).await;
+            loop {
+                let write_ops_since_last_repl = read_txn.qs_read.get_write_ops_since_last_repl();
+                let time_since_last_repl = last_replication_instant.elapsed().as_secs();
+
+                let Some(write_throughput_index) = time_since_last_repl.checked_sub(MIN_REPL_TASK_POLL_INTERVAL) else {
+                    error!("Negative replication window index detected, aborting...");
+                    return;
+                };
+
+                let (current_max_throughput, current_repl_window) = MAX_WRITE_THROUGHPUT_BY_TASK_POLL_INTERVAL[write_throughput_index as usize];
```

Comment:
No check on the bounds here which could cause out of bounds crash. Use .get() instead.

Reply:
wopsies